### PR TITLE
SW-5761 Add subzone selection for observation scheduling

### PR DESCRIPTION
**Pull request by nickgraz**
- Implement new version of web-components with indeterminate checkbox
- Create `ScheduleObservationForm` with zone and subzone rendering and selection

Subsequent PRs:
- Auto check for subzones that have plants
- No plants icon

Depends on https://github.com/terraware/web-components/pull/628

### CODE REVIEW

**Comment by nickgraz** [comment]
* **#2988** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2988?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nickgraz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/scenes/ObservationsRouter/schedule/ObservationSubzoneSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    nextSelectedSubzones.forEach((value: boolean, subzoneId: number) => {
+      if (value) {
+        selectedSubzoneIds.push(subzoneId);
+      }
+    });
+    onChangeSelectedSubzones(selectedSubzoneIds);
+  };
+
+  const onChangeSubzoneCheckbox = (subzoneId: number, value: boolean) => {
+    // Consider using es2015 or above so we can spread iterators and interact with Map a bit better
```
Comment (constanzauanini) [inline src/scenes/ObservationsRouter/schedule/ObservationSubzoneSelector.tsx:43]: Do we have a task to upgrade to es2015?

**Review by constanzauanini — APPROVED** [review]
Lgtm!

**Comment by nickgraz** [comment]
### Merge activity

* **Aug 12, 2:18 PM EDT**: @nickgraz started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2988).

**Pull request merged**